### ISSUE TITLE
`npm run dev` does not quit after receiving Ctrl-C

### ISSUE THREAD

**Issue by zwn** [remix-run/remix#6569]
### What version of Remix are you using?

1.17.0

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

`npm create-remix@latest`
pick Typescript, Cloudflare Pages, no stack, install
`npm run dev`
press Ctrl-C

### Expected Behavior

After printing `Shutting down local server` all subprocess with terminate and control will be returned to the terminal.

### Actual Behavior

It hangs forever. If the process are killed from the outside, the terminal is left in inconsistent state and needs to be reset.

**Comment by zwn** [comment]
Application created from the `cloudflare-workers` template behaves as expected (quits on Ctrl-C and the terminal does not need to be reset).

**Comment by pmbanugo** [comment]
It also happens with Remix 1.16.1. I can't confirm yet if it happens only if you use wrangler 3.

**Comment by pcattori** [comment]
Fixed by #6663 / cannot reproduce on `nightly`:

```sh
npx create-remix@nightly --template https://github.com/remix-run/remix/tree/main/templates/cloudflare-pages
cd my-remix-app
npm run dev
^C
ps aux | rg 'remix' # shows no results
ps aux | rg 'node' # shows no results
```

Tried hitting <kbd>CTRL</kbd><kbd>C</kbd> at different timings (before `wrangler` spins up, after it prints initial messages, after Remix app is running stably, etc...), but nothing I tried resulted in dangling processes.

**Comment by pmbanugo** [comment]
@pcattori  Will this be part of the next release? or only until v2 is shipped?
